### DISCUSSION TITLE
Support for Web Authentication and virtual authenticators

### DISCUSSION THREAD

**Discussion by mudge** [rubycdp/ferrum#422] [Ideas]
We currently make use of [Selenium's Virtual Authenticator](https://www.selenium.dev/documentation/webdriver/interactions/virtual_authenticator/) via [selenium-webdriver's `VirtualAuthenticator`](https://github.com/SeleniumHQ/selenium/blob/240e17b647d754b8d88b606cd1bca6345240c0b1/rb/lib/selenium/webdriver/common/virtual_authenticator/virtual_authenticator.rb) to test adding a hardware security key to our application using [Webauthn](https://www.w3.org/TR/webauthn-2/). Is there a way to expose a similar API in Ferrum?

**Comment by route** [comment]
All is doable, depends on time and human resources. I think it's not going to be my main priority as there are more basic things to finish and implement, but I'll gladly merge any work from you or anyone else on this topic, after all it's same ruby language and OSS so we can copy paste some of the work from other projects.
